roles/test_verify_email/tasks/main.yml

```diff
@@ -0,0 +1,59 @@
+---
+# tasks file for roles/test_verify_email
+
+- name: "Create the alert"
+  ansible.builtin.shell:
+    cmd: |
+      oc apply -f - <<EOF
+      apiVersion: monitoring.coreos.com/v1
+      kind: PrometheusRule
+      metadata:
+        creationTimestamp: null
+        labels:
+          prometheus: default
+          role: alert-rules
+        name: prometheus-alarm-rules
+        namespace: service-telemetry
+      spec:
+        groups:
+          - name: ./openstack.rules
+            rules:
+              - alert: Collectd metrics receive rate is zero
+                expr: rate(sg_total_collectd_msg_received_count[1m]) == 0
+      EOF
+  changed_when: false
+  register: cmd_output
+  failed_when: cmd_output.rc != 0
+
+- name: "Patch the ServiceTelemetry object for the STF deployment"
+  ansible.builtin.shell:
+    cmd: |
+      oc patch stf default --type merge -p '{"spec": {"alertmanagerConfigManifest": "apiVersion: v1\nkind: Secret\nmetadata:\n  name: 'alertmanager-default'\n  namespace: 'service-telemetry'\ntype: Opaque\nstringData:\n  alertmanager.yaml: |-\n    global:\n      resolve_timeout: 10m\n    route:\n      group_by: ['job']\n      group_wait: 30s\n      group_interval: 5m\n      repeat_interval: 12h\n      receiver: 'email'\n    receivers:\n    - name: 'email'\n      - email_configs:\n      - to: mx@redhat.com"}}'
+  changed_when: false
+
+- name: "Interrupt metrics flow by preventing the QDR from running"
+  ansible.builtin.shell:
+    cmd: |
+      for i in {1..30}; do oc delete po -l application=default-interconnect; sleep 1; done
```

Comment:
I think I may be the originator of this technique, and should mention that I don't think it's entirely reliable; I swear I've seen the QDR start up and pass some metrics within 1s, but it's rare.

Here though, I'm a bit extra skeptical because you have ~30s worth of interruption, but the alert query is looking for a rate of 0 metrics over a 60s time frame. I would guess that if there were actually metrics flowing, this interruption would not be long enough to trigger the alert. Since the test is meant to test the email alert function (and related rule/config documentation), I think it's perfectly valid to test it in a quiet system with no metrics and no simulated interruption, but want to point out that we could get false-negatives here for a couple related reasons if the system is not quiet.